---
title: Derleyici Hatası CS1626
ms.date: 07/20/2015
f1_keywords:
- CS1626
helpviewer_keywords:
- CS1626
ms.assetid: 3ba03383-eb24-4fd8-bf40-8b0f7d6baf0d
ms.openlocfilehash: 1263fbd4d6591b1f231e38400c680b549ba5654e
ms.sourcegitcommit: 9b552addadfb57fab0b9e7852ed4f1f1b8a42f8e
ms.translationtype: MT
ms.contentlocale: tr-TR
ms.lasthandoff: 04/23/2019
ms.locfileid: "61686232"
---
# <a name="compiler-error-cs1626"></a><span data-ttu-id="b08d7-102">Derleyici Hatası CS1626</span><span class="sxs-lookup"><span data-stu-id="b08d7-102">Compiler Error CS1626</span></span>
<span data-ttu-id="b08d7-103">Bir catch yan tümcesi ile bir try bloğunun gövdesinde bir değer döndürülemez</span><span class="sxs-lookup"><span data-stu-id="b08d7-103">Cannot yield a value in the body of a try block with a catch clause</span></span>  
  
 <span data-ttu-id="b08d7-104">Try bloğu ile ilişkili bir catch yan tümcesi varsa komutun bir yield deyimi try bloğunda izin verilmiyor.</span><span class="sxs-lookup"><span data-stu-id="b08d7-104">A yield statement is not allowed in a try block if there is a catch clause associated with the try block.</span></span> <span data-ttu-id="b08d7-105">Bu hatayı önlemek için yield deyimi try/catch/finally bloğu dışına taşıyın veya catch bloğu kaldırın.</span><span class="sxs-lookup"><span data-stu-id="b08d7-105">To avoid this error, either move the yield statement out of the try/catch/finally block, or remove the catch block.</span></span>
  
 <span data-ttu-id="b08d7-106">Aşağıdaki örnek, CS1626 oluşturur:</span><span class="sxs-lookup"><span data-stu-id="b08d7-106">The following sample generates CS1626:</span></span>  
  
```csharp  
// CS1626.cs  
using System.Collections;  
  
class C : IEnumerable  
{  
   public IEnumerator GetEnumerator()  
   {  
      try  
      {  
         yield return this;  // CS1626  
      }  
      catch  
      {  
  
      }  
      finally
      {
      
      }
   }  
}  
  
public class CMain  
{  
   public static void Main() { }  
}  
```
